default-token-account/program/src/instruction.rs

```diff
@@ -0,0 +1,135 @@
+//! DefaultTokenAccount program instructions
+
+use solana_program::{
+    instruction::{AccountMeta, Instruction},
+    program_error::ProgramError,
+    pubkey::Pubkey,
+    sysvar,
+};
+
+/// Instructions supported by the DefaultTokenAccount program.
+#[repr(C)]
+#[derive(Clone, Debug, PartialEq)]
+pub enum DefaultTokenAccountInstruction {
+    /// Create the default token account for a user wallet address.
+    /// Fails if the address already exists.
+    ///
+    ///   0. `[writeable]` Default token account address derived from `get_default_token_account_address()`
+    ///   1. `[]` User wallet address (system account)
+    ///   2. `[]` The mint this account will be associated with.
+    ///   3. `[writeable,signer]` The system account that will fund the default token account
+    ///   4. `[]` Rent sysvar
+    ///   5. `[]` System program
+    ///   6. `[]` SPL Token program
+    ///
+    Create,
+
+    /// Asserts that the default token account exists and the provided user wallet address is the
+    /// owner of it.
+    ///
+    ///   0. `[]` Default token account address derived from `get_default_token_account_address()`
+    ///   1. `[]` User wallet address (system account)
+    ///   2. `[]` The mint this account should be associated with.
+    ///   3. `[]` SPL Token program
+    ///
+    Exists,
```

Comment:
From your guide, I was confused by:
> Doing so will cause the `DefaultTokenAccountInstruction::Exists` instruction to fail

and thought that
> Doing so will cause the `DefaultTokenAccountInstruction::Assert` instruction to fail

would be easier to understand

```suggestion
    Assert,
```

Reply:
Ah, sure